src/tzdb.rs

```diff
@@ -0,0 +1,457 @@
+    }
+
+    pub fn from_path<P: AsRef<Path>>(path: P) -> TemporalResult<Self> {
+        tzif::parse_tzif_file(path)
+            .map(Self)
+            .map_err(|e| TemporalError::general(e.to_string()))
+    }
+
+    // There are ultimately
+    pub fn get(&self, epoch_seconds: &Seconds) -> TemporalResult<LocalTimeTypeRecord> {
+        let Some(result) = self.binary_search(epoch_seconds) else {
+            return Err(TemporalError::general("Only Tzif v2+ is supported."));
+        };
+
+        let db = self
+            .0
+            .data_block2
+            .as_ref()
+            .expect("binary search throws error if datablock doesn't exist.");
+
+        match result {
+            Ok(idx) => Ok(get_local_record(db, idx - 1)),
+            Err(idx) if idx == 0 => Ok(get_local_record(db, idx)),
+            Err(idx) => {
+                if db.transition_times.len() <= idx {
+                    return Err(TemporalError::general("TODO: Support POSIX tz string."));
+                }
+                Ok(get_local_record(db, idx - 1))
+            }
+        }
+    }
+
+    // There are various other ways to search rather than binary_search. See glibc
+    pub fn binary_search(&self, epoch_seconds: &Seconds) -> Option<Result<usize, usize>> {
+        self.0
+            .data_block2
+            .as_ref()
+            .map(|b| b.transition_times.binary_search(epoch_seconds))
+    }
+
+    pub fn v2_estimate_tz_pair(
+        &self,
+        seconds: &Seconds,
+    ) -> TemporalResult<Vec<LocalTimeTypeRecord>> {
+        // We need to estimate a tz pair.
+        // First search the ambiguous seconds.
+        // TODO: it would be nice to resolve the Posix str into a local time type record.
+        let Some(b_search_result) = self.binary_search(seconds) else {
+            return Err(TemporalError::general("Only Tzif v2+ is supported."));
+        };
+
+        let data_block = self
+            .0
+            .data_block2
+            .as_ref()
+            .expect("binary_search validates that data_block2 exists.");
+
+        let estimated_idx = match b_search_result {
+            Ok(idx) => idx,
+            Err(idx) if idx == 0 => return Ok(vec![get_local_record(data_block, idx)]),
+            Err(idx) => {
+                if data_block.transition_times.len() <= idx {
+                    return Err(TemporalError::general("TODO: Support POSIX tz string."));
+                }
+                idx
+            }
+        };
+
+        // The estimated index will be off based on the amount missing
+        // from the lack of offset.
+        //
+        // This means that we may need (idx, idx - 1) or (idx - 1, idx - 2)
+        let record = get_local_record(data_block, estimated_idx);
+        let record_minus_one = get_local_record(data_block, estimated_idx - 1);
+
+        // Potential shift bugs with odd historical transitions?
+        let shift_window = usize::from((record.utoff + record_minus_one.utoff).0.signum() >= 0);
```

Comment:
It took me a bit of time to understand that this is just checking if the sum of offsets is positive, so maybe just do that?
```suggestion
let shift_window = usize::from(record.utoff + record_minus_one.utoff >= Seconds(0))
```